### PR TITLE
PLAT-1081-4 IDisplayString: Implement neither Comparable, nor hashCode, nor equals

### PR DESCRIPTION
**Pull request by alexander-schmidt**

### CODE REVIEW

**Review by daniel-klose — APPROVED** [review]

File: platform-common/src/main/java/com/softicar/platform/common/core/i18n/IDisplayString.java [hunk 1]
```diff
@@ -12,16 +12,14 @@
...
  * @author Oliver Richers
  */
-public interface IDisplayString extends Comparable<IDisplayString> {
+public interface IDisplayString {
 
 	/**
-	 * An immutable, empty {@link IDisplayString}.
+	 * An empty {@link IDisplayString}.
 	 */
 	IDisplayString EMPTY = PlainDisplayString.getEmpty();
```
Comment (oliver-richers) [inline platform-common/src/main/java/com/softicar/platform/common/core/i18n/IDisplayString.java:24]: IMO, there is not much use in this getEmpty() method, we could remove it
```suggestion
	IDisplayString EMPTY = new PlainDisplayString("");
```

File: platform-dom/src/test/java/com/softicar/platform/dom/elements/AbstractDomValueSelectTest.java [hunk 2]
```diff
@@ -126,7 +126,7 @@ private void assertDisplayNames(List<Pair<Integer, IDisplayString>> values, Stri
 			if (displayNames[i] != null) {
 				IDisplayString expected = IDisplayString.create(displayNames[i]);
 				IDisplayString actual = values.get(i).getSecond();
-				assertTrue(expected.compareTo(actual) == 0);
+				assertTrue(expected.toString().compareTo(actual.toString()) == 0);
```
Comment (oliver-richers) [inline platform-dom/src/test/java/com/softicar/platform/dom/elements/AbstractDomValueSelectTest.java:129]:
```suggestion
				assertEquals(expected.toString(), actual.toString());
```
maybe add an `assertEquals(IDisplayString,IDisplayString)` to `Asserts`?

File: platform-emf/src/main/java/com/softicar/platform/emf/matrix/EmfSettingMatrixView.java [hunk 3]
```diff
@@ -327,7 +327,7 @@ private <X> Collection<X> sort(Collection<X> collection, Function<X, IDisplayStr
 
 			return collection//
 				.stream()
-				.sorted(Comparator.<X, IDisplayString> comparing(displayNamer::apply))
+				.sorted(Comparator.comparing(x -> displayNamer.apply(x).toString()))
```
Comment (oliver-richers) [inline platform-emf/src/main/java/com/softicar/platform/emf/matrix/EmfSettingMatrixView.java:330]: `displayNamer`? really? ;)

**Review by oliver-richers — CHANGES_REQUESTED** [review]

File: platform-common/src/main/java/com/softicar/platform/common/core/i18n/IDisplayString.java (hunk 1 above)
Comment (alexander-schmidt) [reply platform-common/src/main/java/com/softicar/platform/common/core/i18n/IDisplayString.java:24]: indeed

File: platform-dom/src/test/java/com/softicar/platform/dom/elements/AbstractDomValueSelectTest.java (hunk 2 above)
Comment (alexander-schmidt) [reply platform-dom/src/test/java/com/softicar/platform/dom/elements/AbstractDomValueSelectTest.java:129]: added that

File: platform-emf/src/main/java/com/softicar/platform/emf/matrix/EmfSettingMatrixView.java (hunk 3 above)
Comment (alexander-schmidt) [reply platform-emf/src/main/java/com/softicar/platform/emf/matrix/EmfSettingMatrixView.java:330]: uhm... yea. admittedly, slightly odd - but short.

**Review by oliver-richers — APPROVED** [review]

**Pull request merged**